lib/Auth/Process/COmanageDbClient.php

```diff
@@ -323,8 +324,20 @@ public function process(&$state)
                   $callback = SimpleSAML_Module::getModuleURL('attrauthcomanage/idp_callback.php', array('stateId' => $id));
                   SimpleSAML_Logger::debug("[attrauthcomanage] process: callback url => " . $callback);
                   $params = array("targetnew" => $callback);
+                  // Check if community signup is required
                   if (!empty($state['saml:AuthenticatingAuthority']) && in_array(end($state['saml:AuthenticatingAuthority']), $this->communityIdps, true)) {
-                    \SimpleSAML\Utils\HTTP::redirectTrustedURL($this->registryUrls['community_sign_up'], $params);
+                      // Redirect to community signup flow with all
+                      // attributes available including affiliation
+                      if (!empty($state['Attributes']['voPersonExternalAffiliation'])
+                          && !empty($state['Attributes']['mail'])
+                          && !empty($state['Attributes']['givenName'])
+                          && !empty($state['Attributes']['sn'])) {
+                          \SimpleSAML\Utils\HTTP::redirectTrustedURL($this->registryUrls['community_sign_up'], $params);
+                      } elseif (!empty($this->registryUrls['community_sign_up_no_aff'])) {
+                          \SimpleSAML\Utils\HTTP::redirectTrustedURL($this->registryUrls['community_sign_up_no_aff'], $params);
+                      } else {
+                          \SimpleSAML\Utils\HTTP::redirectTrustedURL($this->registryUrls['community_sign_up'], $params);
+                      }
                   }
```

Comment:
This can be simplified as follows:
```
if (!empty($state['saml:AuthenticatingAuthority']) && in_array(end($state['saml:AuthenticatingAuthority']), $this->communityIdps, true)) {

    if (empty($this->registryUrls['community_sign_up_no_aff']) || 
        (!empty($state['Attributes']['voPersonExternalAffiliation'])
        && !empty($state['Attributes']['mail'])
        && !empty($state['Attributes']['givenName'])
        && !empty($state['Attributes']['sn']))) {
        // Redirect to default community signup flow if 
        // 1. there is no other specific community signup defined
        // or
        // 2. all signup attributes are available, including affiliation
        \SimpleSAML\Utils\HTTP::redirectTrustedURL($this->registryUrls['community_sign_up'], $params);
    } else {
        \SimpleSAML\Utils\HTTP::redirectTrustedURL($this->registryUrls['community_sign_up_no_aff'], $params);
    }
}
```